roomserver/api/query.go

```diff
@@ -203,6 +203,8 @@ type QueryStateAndAuthChainRequest struct {
 	PrevEventIDs []string `json:"prev_event_ids"`
 	// The list of auth events for the event. Used to calculate the auth chain
 	AuthEventIDs []string `json:"auth_event_ids"`
+	// Should state resolution be ran on the result events?
+	ResolveState bool `json:"resolve_state"`
```

Comment:
When would we *not* want to run state resolution?

Reply:
Without examining things that call `QueryStateAndAuthChain` I am not sure, so I've taken the non-breaking approach for now.

Reply:
Please flag it as such then e.g `// TODO: check call sites and remove if we always want to do state res`

Reply:
Done.